toplevel/vernacentries.ml

```diff
@@ -136,7 +136,10 @@ let make_cases_aux glob_ref =
 		   let n' = Namegen.next_name_away_with_default (Id.to_string Namegen.default_dependent_ident) n avoid in
 		   Id.to_string n' :: rename (n'::avoid) l in
 	     let al' = rename [] al in
-	     (Id.to_string consname :: al') :: l)
+	     let dirpath = Nametab.dirpath_of_global glob_ref in
+	     let consname_qualid = Libnames.make_qualid dirpath consname in
+	     let consname_global = Nametab.locate consname_qualid in
+	     (Libnames.string_of_qualid (Nametab.shortest_qualid_of_global Id.Set.empty consname_global) :: al') :: l)
```

Comment:
Thanks! Extremely useful information. @psteckler do you think you can incorporate some of Hugo's feedback into the patch? Otherwise maybe you could write his comments down in as an OCaml comment for future hacking.